src/cli/self_update/windows.rs

```diff
@@ -84,6 +84,93 @@ pub(crate) fn choose_vs_install(process: &Process) -> Result<Option<VsInstallPla
     Ok(plan)
 }
 
+pub(super) async fn maybe_install_msvc(
+    term: &mut ColorableTerminal,
+    no_prompt: bool,
+    quiet: bool,
+    opts: &InstallOpts<'_>,
+    process: &Process,
+) -> Result<()> {
+    let Some(plan) = do_msvc_check(opts, process) else {
+        return Ok(());
+    };
+
+    if no_prompt {
+        warn!("installing msvc toolchain without its prerequisites");
+    } else if !quiet && plan == VsInstallPlan::Automatic {
+        md(term, MSVC_AUTO_INSTALL_MESSAGE);
+        match choose_vs_install(process)? {
+            Some(VsInstallPlan::Automatic) => {
+                match try_install_msvc(opts, process).await {
+                    Err(e) => {
+                        // Make sure the console doesn't exit before the user can
+                        // see the error and give the option to continue anyway.
+                        report_error(&e, process);
+                        if !common::question_bool("\nContinue?", false, process)? {
+                            info!("aborting installation");
+                        }
+                    }
+                    Ok(ContinueInstall::No) => ensure_prompt(process)?,
+                    _ => {}
+                }
+            }
+            Some(VsInstallPlan::Manual) => {
+                md(term, MSVC_MANUAL_INSTALL_MESSAGE);
+                if !common::question_bool("\nContinue?", false, process)? {
+                    info!("aborting installation");
+                }
+            }
+            None => {}
+        }
+    } else {
+        md(term, MSVC_MESSAGE);
+        md(term, MSVC_MANUAL_INSTALL_MESSAGE);
+        if !common::question_bool("\nContinue?", false, process)? {
+            info!("aborting installation");
+        }
+    }
+
+    Ok(())
+}
+
+#[cfg(windows)]
```

Comment:
The whole module is under that cfg now so it makes sense to remove these in this particular file:

```suggestion
```

Reply:
Yup!